rsocket/RSocketClient.cpp

```diff
@@ -129,6 +129,10 @@ void RSocketClient::disconnect(folly::exception_wrapper ex) {
   });
 }
 
+bool RSocketClient::isDisconnectedOrClosed() const {
```

Comment:
This is not a thread safe method. Do you need it in the RSocketClient class?
The StreamsFactory is using the RSocketStateMachine:: isDisconnectedOrClosed, is that correct?